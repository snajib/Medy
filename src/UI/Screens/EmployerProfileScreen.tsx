import React from "react";
import { Gradient } from "../Shared/linear-gradient";
import { View, Text } from "react-native";
import { styles } from "../Shared/styles";
import { Accordion } from "../Shared/Accordion/accordion";
import { ScrollView } from "react-native-gesture-handler";
import { profileOptions } from "../Shared/Accordion/accordion-schemas"

export class EmployerProfileScreen extends React.Component {
  render() {
    return (
      <View style={styles.topPrimaryContainterStyle}>
        <Gradient></Gradient>
        <View style={styles.topSecondaryContainerStyle}>
          <Text style={styles.titleText}>Profile</Text>
          <View style={styles.wholeDefaultCardStyle}>
            <ScrollView>
              <Accordion schema={profileOptions}></Accordion>
            </ScrollView>
          </View>
        </View>
      </View>
    );
  }
}
