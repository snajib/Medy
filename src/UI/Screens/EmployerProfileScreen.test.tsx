import React from 'react';
import renderer from 'react-test-renderer';
import { EmployerProfileScreen } from './EmployerProfileScreen';
import { View, Text } from 'react-native';

describe('EmployerProfileScreen', () => {
  it('should render header correctly', () => {
    const profileScreen = renderer.create(<EmployerProfileScreen />);

    expect(profileScreen.root.findAllByType(Text).length).toBe(7);
    expect(profileScreen.root.findAllByType(Text)[0].props.children).toBe(
      'Profile',
    );
  });

  it('should render views correctly', () => {
    const profileScreen = renderer.create(<EmployerProfileScreen />);

    expect(profileScreen.root.findAllByType(View).length).toBe(17);
  });

  it('should render with snapshot', () => {
    const employerProfScreen = renderer.create(<EmployerProfileScreen />);
    expect(employerProfScreen.toJSON()).toMatchSnapshot();
  });
});
