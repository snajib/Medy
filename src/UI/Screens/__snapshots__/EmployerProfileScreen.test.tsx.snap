// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployerProfileScreen should render with snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5F44BA",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": "22%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 36,
          "fontStyle": "normal",
          "fontWeight": "bold",
          "paddingLeft": "3.5%",
          "textAlign": "left",
        }
      }
    >
      Profile
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 12,
          "height": 1334,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 10,
          "width": 750,
        }
      }
    >
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "paddingLeft": "5%",
                "paddingTop": "2.5%",
              }
            }
          >
            <View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                      "fontStyle": "normal",
                      "fontWeight": "bold",
                      "paddingTop": "5%",
                      "textAlign": "left",
                    }
                  }
                >
                  Company Pictures
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": 0,
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <Text>
                    accordion content
                  </Text>
                </View>
              </View>
            </View>
            <View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                      "fontStyle": "normal",
                      "fontWeight": "bold",
                      "paddingTop": "5%",
                      "textAlign": "left",
                    }
                  }
                >
                  About
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": 0,
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <Text>
                    accordion content
                  </Text>
                </View>
              </View>
            </View>
            <View>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                      "fontStyle": "normal",
                      "fontWeight": "bold",
                      "paddingTop": "5%",
                      "textAlign": "left",
                    }
                  }
                >
                  Technologies
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": 0,
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <Text>
                    accordion content
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
